### PR TITLE
fix custom url issues

### PR DESCRIPTION
**Pull request by tharindu1st**

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/settings/settings.jag [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<%
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Rajith90) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/settings/settings.jag]:
```suggestion
 * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js [hunk 2]
```diff
@@ -20,9 +20,73 @@
...
 var getLoopbackOrigin = function() {
     var mgtTransportPort = utils.getCarbonTransportPort("https");
     var origin = 'https://' + app.origin.host + ":" + mgtTransportPort;
     return origin;
 };
+
+var getTenantBaseStoreContext = function() {
+    var tenantDomain = getTenantDomain();
+    var tenantContext = utils.getTenantBasedDevPortalContext(tenantDomain);
+    if(tenantContext != null){
```
Comment (Rajith90) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js]:
```suggestion
    if (tenantContext != null) {
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js [hunk 3]
```diff
@@ -20,9 +20,73 @@
...
+        return tenantContext
+    }else{
```
Comment (Rajith90) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js]:
```suggestion
    } else {
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js [hunk 4]
```diff
@@ -20,9 +20,73 @@
...
+        return app.context;
+    }
+};
+
+var getTenantBasedLoginCallBack = function() {
+    var tenantDomain = getTenantDomain();
+    var storeDomainMapping = utils.getTenantBasedStoreDomainMapping(tenantDomain);
+    if(storeDomainMapping != null){
+       if(storeDomainMapping.get('login') != null){
```
Comment (Rajith90) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js]:
```suggestion
       if (storeDomainMapping.get('login') != null) {
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js [hunk 5]
```diff
@@ -20,9 +20,73 @@
...
+        return tenantContext
+    }else{
+        return app.context;
+    }
+};
+
+var getTenantBasedLoginCallBack = function() {
+    var tenantDomain = getTenantDomain();
+    var storeDomainMapping = utils.getTenantBasedStoreDomainMapping(tenantDomain);
+    if(storeDomainMapping != null){
```
Comment (Rajith90) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js]:
```suggestion
    if (storeDomainMapping != null) {
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js [hunk 6]
```diff
@@ -20,9 +20,73 @@
...
+       }
+        return "https://"+storeDomainMapping.get('customUrl')+LOGIN_CALLBACK_URL_SUFFIX;
+    }else{
+    return null;
+    }
+};
+var getTenantBasedLogoutCallBack = function() {
+    var tenantDomain = getTenantDomain();
+    var storeDomainMapping = utils.getTenantBasedStoreDomainMapping(tenantDomain);
+    if(storeDomainMapping != null){
```
Comment (Rajith90) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js]:
```suggestion
    if (storeDomainMapping != null) {
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js [hunk 7]
```diff
@@ -20,9 +20,73 @@
...
+       if(storeDomainMapping.get('logout') != null){
```
Comment (Rajith90) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js]:
```suggestion
       if (storeDomainMapping.get('logout') != null) {
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js [hunk 8]
```diff
@@ -20,9 +20,73 @@
...
+        return storeDomainMapping.get('logout');
+       }
+        return "https://"+storeDomainMapping.get('customUrl')+LOGOUT_CALLBACK_URL_SUFFIX;
+    }else{
```
Comment (Rajith90) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js]:
```suggestion
    } else {
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js [hunk 9]
```diff
@@ -20,9 +20,73 @@
...
+        if(tenantDomain == null){
+            tenantDomain = "carbon.super";
+        }
+    }
+    return tenantDomain;
+};
+
+var getServiceProviderTenantDomain = function(){
+    var tenantDomain = getTenantDomain();
+    if(isPerTenantServiceProviderEnabled()){
```
Comment (Rajith90) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js]:
```suggestion
    if (isPerTenantServiceProviderEnabled()) {
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js [hunk 10]
```diff
@@ -20,9 +20,73 @@
...
+        return tenantDomain;
+    }else{
```
Comment (Rajith90) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js]:
```suggestion
    } else {
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js [hunk 11]
```diff
@@ -20,9 +20,73 @@
...
+    var tenantDomain = getTenantDomain();
+
+    var perTenantServiceProviderEnabled = utils.isPerTenantServiceProviderEnabled(tenantDomain);
+    return perTenantServiceProviderEnabled;
+};
+
+var getTenantDomain = function(){
+    
+    var tenantDomain = request.getParameter("tenant");
+    if(tenantDomain == null){
```
Comment (Rajith90) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js]:
```suggestion
    if (tenantDomain == null) {
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js [hunk 12]
```diff
@@ -20,9 +20,73 @@
...
+        tenantDomain = request.getHeader("X-WSO2-Tenant");
+        if(tenantDomain == null){
```
Comment (Rajith90) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js]:
```suggestion
        if (tenantDomain == null) {
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/refresh/refresh.jag [hunk 13]
```diff
@@ -20,7 +20,11 @@
     include("/services/constants.jag");
     var appUtils = require("/services/utils.js");
     var app = require("/site/public/theme/settings.js").Settings.app;
-
+    app.context = appUtils.getTenantBaseStoreContext();
+    var tenantDomain;
+     if(appUtils.isPerTenantServiceProviderEnabled()){
```
Comment (Rajith90) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/refresh/refresh.jag]:
```suggestion
     if (appUtils.isPerTenantServiceProviderEnabled()) {
```

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java [hunk 14]
```diff
@@ -5463,7 +5463,8 @@ public String getOpenAPIDefinitionForEnvironment(Identifier apiId, String enviro
             //todo: use get api by id, so no need to set scopes or uri templates
             api.setScopes(oasParser.getScopes(definition));
             api.setUriTemplates(oasParser.getURITemplates(definition));
-            apiTenantDomain = MultitenantUtils.getTenantDomain(api.getId().getProviderName());
+            apiTenantDomain = MultitenantUtils.getTenantDomain(
```
[the remark this answers is not in the document]
Comment (isharac) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java]: As offline checked this is fine as provider name contains the @carbon.super

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/services/login/idp.jag [hunk 15]
```diff
@@ -90,7 +90,7 @@
                 var clientSecret = result.data.clientSecret;
 
                 var addApplicationKey = systemApplicationDAO
-                                                .addApplicationKey(PUBLISHER_CLIENT_APP_NAME, clientId, clientSecret);
+                                                .addApplicationKey(PUBLISHER_CLIENT_APP_NAME, clientId, clientSecret,SUPER_TENANT_DOMAIN);
```
Comment (isharac) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/services/login/idp.jag]: space required.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/services/login/login_callback.jag [hunk 16]
```diff
@@ -40,9 +40,9 @@
     var SystemApplicationDTO = Packages.org.wso2.carbon.apimgt.impl.dao.SystemApplicationDAO;
     var systemApplicationDAO = new SystemApplicationDTO();
     // this is to support migration from admin_publisher to apim_publisher
-    var systemApplicationDTO = systemApplicationDAO.getClientCredentialsForApplication(PUBLISHER_CLIENT_APP_NAME);
+    var systemApplicationDTO = systemApplicationDAO.getClientCredentialsForApplication(PUBLISHER_CLIENT_APP_NAME,SUPER_TENANT_DOMAIN);
```
Comment (isharac) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/services/login/login_callback.jag]: space is required. please check the other places as well.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.dcr/src/main/java/org/wso2/carbon/apimgt/rest/api/dcr/web/impl/RegistrationServiceImpl.java [hunk 17]
```diff
@@ -137,8 +140,11 @@ public Response register(RegistrationProfile profile) {
                 oauthApplicationInfo.setIsSaasApplication(profile.isSaasApp());
                 oauthApplicationInfo.setTokenType(tokenType);
                 appRequest.setOAuthApplicationInfo(oauthApplicationInfo);
-
-                loggedInUserTenantDomain = RestApiUtil.getLoggedInUserTenantDomain();
+                if (!authUserName.equals(owner)){
+                    loggedInUserTenantDomain = MultitenantUtils.getTenantDomain(owner);
+                }else{
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.dcr/src/main/java/org/wso2/carbon/apimgt/rest/api/dcr/web/impl/RegistrationServiceImpl.java]:
```suggestion
                } else {
```

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 18]
```diff
@@ -7919,7 +7960,7 @@ public static boolean isEnabledSubscriptionSpikeArrest() {
                 if (policy instanceof SubscriptionPolicy) {
                     SubscriptionPolicy subscriptionPolicy = (SubscriptionPolicy) policy;
                     setBillingPlanAndCustomAttributesToTier(subscriptionPolicy, tier);
-                    if(StringUtils.equals(subscriptionPolicy.getBillingPlan(),APIConstants.COMMERCIAL_TIER_PLAN)){
+                    if(StringUtils.equals(subscriptionPolicy.getBillingPlan(), APIConstants.COMMERCIAL_TIER_PLAN)){
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]:
```suggestion
                    if (StringUtils.equals(subscriptionPolicy.getBillingPlan(), APIConstants.COMMERCIAL_TIER_PLAN)) {
```

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 19]
```diff
@@ -9936,5 +9981,103 @@ public static boolean validateAPICategories(List<APICategory> categories, String
...
+            throw new APIManagementException(msg, e);
+        } catch (ClassCastException e) {
+            String msg = "Invalid JSON found in the gateway tenant domain mappings";
+            log.error(msg, e);
+            throw new APIManagementException(msg, e);
+        } catch (ParseException e) {
+            String msg = "Malformed JSON found in the gateway tenant domain mappings";
+            log.error(msg, e);
+            throw new APIManagementException(msg, e);
+        }
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: Shall we remove logging the errors? Throwing the error only sufficient.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 20]
```diff
@@ -9936,5 +9981,103 @@ public static boolean validateAPICategories(List<APICategory> categories, String
...
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: Same as above.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 21]
```diff
@@ -9936,5 +9981,103 @@ public static boolean validateAPICategories(List<APICategory> categories, String
...
+    public static boolean isPerTenantServiceProviderEnabled(String tenantDomain) throws APIManagementException {
+
+        int tenantId = getTenantIdFromTenantDomain(tenantDomain);
+        JSONObject tenantConfig = getTenantConfig(tenantId);
+        if (tenantConfig.containsKey(APIConstants.PER_TENANT_SERVICE_PROVIDER)) {
+            return (boolean) tenantConfig.get(APIConstants.PER_TENANT_SERVICE_PROVIDER);
+        }
+        return false;
+    }
+    public static String getTenantAdminUserName(String tenantDomain) throws APIManagementException {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]:
```suggestion

    public static String getTenantAdminUserName(String tenantDomain) throws APIManagementException {
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/jaggery.conf [hunk 22]
```diff
@@ -60,6 +60,10 @@
             "url": "/settings/*",
             "path": "/site/public/pages/index.jag"
         },
+        {
+        "url":"/services/settings/settings.js",
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/jaggery.conf]:
```suggestion
            "url":"/services/settings/settings.js",
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/jaggery.conf [hunk 23]
```diff
@@ -60,6 +60,10 @@
...
+        "path":"/services/settings/settings.jag"
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/jaggery.conf]:
```suggestion
            "path":"/services/settings/settings.jag"
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/login/idp.jag [hunk 24]
```diff
@@ -40,8 +40,14 @@
     var settingsResponse = get(settingsAPIUrl, {}, "json");
 
     var dcrUrl = appUtils.getLoopbackOrigin() + DCR_URL_SUFFIX;
-    var loginCallbackUrl = serverUrl + app.context + LOGIN_CALLBACK_URL_SUFFIX;
-    var logoutCallbackUrl = serverUrl + app.context + LOGOUT_CALLBACK_URL_SUFFIX;
+    var loginCallbackUrl = appUtils.getTenantBasedLoginCallBack();
+    if(loginCallbackUrl == null){
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/login/idp.jag]:
```suggestion
    if (loginCallbackUrl == null) {
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/login/idp.jag [hunk 25]
```diff
@@ -40,8 +40,14 @@
...
+        loginCallbackUrl = serverUrl + app.context + LOGIN_CALLBACK_URL_SUFFIX;
+    }
+    var logoutCallbackUrl = appUtils.getTenantBasedLogoutCallBack();
+    if(logoutCallbackUrl == null){
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/login/idp.jag]:
```suggestion
    if (logoutCallbackUrl == null) {
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/login/idp.jag [hunk 26]
```diff
@@ -40,8 +40,14 @@
...
+         logoutCallbackUrl = serverUrl + app.context + LOGOUT_CALLBACK_URL_SUFFIX;
+     }
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/login/idp.jag]:
```suggestion
    }
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/login/idp.jag [hunk 27]
```diff
@@ -59,32 +65,35 @@
...
                 var dcrRequestData = {
                     "callbackUrl": callbackUrl,
                     "clientName": STORE_CLIENT_APP_NAME,
-                    "owner": String(utils.getAdminUsername()),
+                    "owner": String(utils.getTenantAdminUserName(serviceProviderTenantDomain)),
                     "grantType": "authorization_code refresh_token",
                     "saasApp": true
                 };
-
+                log.info(dcrRequestData);
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/login/idp.jag]: Shall we remove this?

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/login/login_callback.jag [hunk 28]
```diff
@@ -28,17 +29,25 @@
     } else {
         serverUrl = utils.getServerURL();
     }
+    var loginCallbackUrl = appUtils.getTenantBasedLoginCallBack();
+    if(loginCallbackUrl == null){
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/login/login_callback.jag]:
```suggestion
    if (loginCallbackUrl == null) {
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/login/login_callback.jag [hunk 29]
```diff
@@ -28,17 +29,25 @@
...
+        loginCallbackUrl = serverUrl + app.context + LOGIN_CALLBACK_URL_SUFFIX;
+     }
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/login/login_callback.jag]:
```suggestion
    }
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/login/login_callback.jag [hunk 30]
```diff
@@ -28,17 +29,25 @@
...
     var tokenRequestData = {
         "grant_type": "authorization_code",
         "code": request.getParameter("code","UTF-8"),
-        "redirect_uri": serverUrl + app.context + LOGIN_CALLBACK_URL_SUFFIX
+        "redirect_uri": loginCallbackUrl
     };
+     var tenantDomain = "carbon.super" ;
+     if(appUtils.isPerTenantServiceProviderEnabled()){
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/login/login_callback.jag]:
```suggestion
     if (appUtils.isPerTenantServiceProviderEnabled()) {
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/login/login_callback.jag [hunk 31]
```diff
@@ -28,17 +29,25 @@
...
+        tenantDomain = appUtils.getTenantDomain();
+      }
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/login/login_callback.jag]:
```suggestion
     }
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/logout/logout.jag [hunk 32]
```diff
@@ -38,7 +39,10 @@
     if (idpConfig != null) {
         logoutEndpoint = idpConfig.getOidcLogoutEndpoint();
     }
-    var postLogoutRedirectURI = serverUrl + app.context + LOGOUT_CALLBACK_URL_SUFFIX;
+    var postLogoutRedirectURI = appUtils.getTenantBasedLogoutCallBack();
+    if(postLogoutRedirectURI == null){
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/logout/logout.jag]:
```suggestion
    if (postLogoutRedirectURI == null) {
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/logout/logout.jag [hunk 33]
```diff
@@ -38,7 +39,10 @@
...
+         postLogoutRedirectURI = serverUrl + app.context + LOGOUT_CALLBACK_URL_SUFFIX;
+     }
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/logout/logout.jag]:
```suggestion
    }
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/refresh/refresh.jag (hunk 13 above)
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/refresh/refresh.jag]:
```suggestion
    if (appUtils.isPerTenantServiceProviderEnabled()) {
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/refresh/refresh.jag [hunk 34]
```diff
@@ -20,7 +20,11 @@
...
+        tenantDomain = appUtils.getTenantDomain();
+      }
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/refresh/refresh.jag]:
```suggestion
     }
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js [hunk 35]
```diff
@@ -20,9 +20,73 @@
...
+    if (storeDomainMapping != null) {
+       if (storeDomainMapping.get('login') != null) {
+        return storeDomainMapping.get('login');
+       }
+        return "https://"+storeDomainMapping.get('customUrl')+LOGIN_CALLBACK_URL_SUFFIX;
+    }else{
+    return null;
+    }
+};
+var getTenantBasedLogoutCallBack = function() {
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/services/utils.js]:
```suggestion

var getTenantBasedLogoutCallBack = function() {
```

**Pull request merged**